packages/@sanity/default-layout/src/navbar/search/contexts/search/reducer.ts

```diff
@@ -218,12 +240,31 @@ export function searchReducer(state: SearchReducerState, action: SearchAction):
           ...state.result,
           loaded: false,
         },
-        terms: {
+        terms: stripRecentIndexAndTimestamp({
           ...state.terms,
           types: [],
-        },
+        }),
       }
     default:
       return state
   }
 }
+
+/**
+ * This function is used to strip both __index and __recentTimestamp from terms, generally whenever
+ * there's a change in search* terms or options that would otherwise trigger an additional search request.
+ * (e.g. updating the search query, changing a sort filter, adding / removing document types)
+ *
+ * This is done so we can better disambiguate between requests sent as a result of clicking a 'recent search'
+ * for purposes of measurement. __index is used to measure exactly which recent search item was clicked.
+ *
+ * @todo remove this (and associated tests) once client-side instrumentation is available
+ */
+function stripRecentIndexAndTimestamp(terms: RecentSearchTerms | SearchTerms) {
+  if (isRecentSearchTerms(terms)) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const {__index, __recentTimestamp, ...rest} = terms
```

Comment:
I think there is a case to be made for putting these in a wrapper object, so we only need to check for one property. If more of these get added in the future, it is likely that this callsite gets forgotten about.